packages/flutter/lib/src/material/refresh_indicator.dart

```diff
@@ -261,8 +261,14 @@ class RefreshIndicatorState extends State<RefreshIndicator> with TickerProviderS
     super.dispose();
   }
 
+  // The [dragDetails] is non-null if the scroll notifications are triggered by dragging.
+  // The indicator can only be triggered by dragging, and cannot be triggered
+  // by other non-dragging behaviors such as [ScrollController.jumpTo].
+  // See also [RefreshIndicator.triggerMode].
   bool _shouldStart(ScrollNotification notification) {
-    return (notification is ScrollStartNotification || (notification is ScrollUpdateNotification && notification.dragDetails != null && widget.triggerMode == RefreshIndicatorTriggerMode.anywhere))
+    return ((notification is ScrollStartNotification && notification.dragDetails != null) // Start dragging at zero scroll position.
```

Comment:
This comment is a bit confusing, I think what I am looking for is why we need to check notification.dragDetails != null, This seems to be quite convoluted.

I assume something the explanation will be something like this.

```
If the notification.dragDetails is null, this scroll is not triggered by user dragging. It may be a result of ScrollController.jumpTo. In this case we don't want to trigger the refresh indicator.
```